os/provisioning.md

```diff
@@ -0,0 +1,127 @@
+  discovery:                   https://discovery.etcd.io/<token>
+```
+
+But what are `PUBLIC_IPV4` and `PRIVATE_IPV4`? These variables are populated from the environment in which Container Linux runs (e.g. EC2, Azure, GCE), assuming this metadata exists. In the default EC2 case, the `public_ipv4` and `local_ipv4` metadata would be used. On bare metal, this information cannot be reliably derived in a general manner, so these variables couldn't be used. In Azure, *either* the virtual IP or public IP could be used for the `PUBLIC_IPV4` and the dynamic IP would be used for the `PRIVATE_IPV4`. These are just a few examples. Given the many different environments in which Container Linux can run, it's difficult if not impossible to accurately determine these variables in every scenario.
+
+On top of this variable-expansion problem, it is very common for users to inadvertently write invalid configs. Instead of forcing users to waste time testing potentially invalid configs, the use of a transformation tool is instead encouraged. The default tool recommended for this task is the [Config Transpiler][ct] (ct for short). It is responsible for validating and transforming a Container Linux Config into the format that Container Linux can consume: the Ignition Config.
+
+[clc]: https://coreos.com/os/docs/latest/configuration.html
+[ct]: https://github.com/coreos/container-linux-config-transpiler/releases
+
+### Ignition Config
+
+The Ignition Config is the configuration format consumed directly by Container Linux. Ignition, the utility inside of Container Linux which is responsible for provisioning the machine, actually fetches and executes the Ignition Config.
+
+Ignition Configs are mostly static, distro-agnostic, and meant to be generated by a machine rather than a human. While they can be written directly by users, it is highly discouraged due to the easy that which errors can be introduced. Instead of writing Ignition Configs directly, users are encouraged to use provisioning tools like Matchbox, which transparently translate Container Linux Configs to Ignition Configs, or use the Config Transpiler directly.
+
+![visual overview of the alternate ct workflows](img/transpile-workflow.svg)
+
+As can be seen above, ct will typically only be manually invoked when users are manually provisioning machines. If a provisioning tool like Matchbox is used, ct will transparently be incorporated into the deployment pipeline. In this scenario, the end user only needs to worry about preparing a Container Linux Config - Ignition and the Ignition are merely an implementation detail and knowledge of their existence isn't necessary.
+
+## Config Transpiler
+
+The Container Linux Config Transpiler is the tool that makes it possible for the end user to remain distanced from the details of configuring Container Linux. It's responsible for transforming a Container Linux Config written by an end user into an Ignition Config to be consumed by instances of Container Linux.
```

Comment:
FIrst sentence would be better maybe like "The Container Linux Config Transpiler abstracts the details of configuring Container Linux."
s/end user/user/ (or "human", even) (throughout this line)